Make.inc

```diff
@@ -562,34 +584,38 @@ endif
 endif
 
 ifeq ($(OS),WINNT)
-LIBUNWIND=
+LIBUNWIND:=
 else
 ifeq ($(USE_SYSTEM_LIBUNWIND), 1)
 ifneq ($(OS),Darwin)
-LIBUNWIND=-lunwind-generic -lunwind
+LIBUNWIND:=-lunwind-generic -lunwind
 endif
 else
 ifeq ($(OS),Darwin)
-LIBUNWIND=$(build_libdir)/libosxunwind.a
+LIBUNWIND:=$(build_libdir)/libosxunwind.a
 JCPPFLAGS+=-DLIBOSXUNWIND
 else
-LIBUNWIND=$(build_libdir)/libunwind-generic.a $(build_libdir)/libunwind.a
+LIBUNWIND:=$(build_libdir)/libunwind-generic.a $(build_libdir)/libunwind.a
 endif
 endif
 endif
 
 ifeq ($(USE_SYSTEM_LLVM), 1)
-LLVM_CONFIG ?= llvm-config$(EXE)
+LLVM_CONFIG ?= llvm-config$(HOST_EXE)
 JCPPFLAGS+=-DSYSTEM_LLVM
-LLVM_VER = $(shell $(LLVM_CONFIG) --version)
+LLVM_VER := $(shell $(LLVM_CONFIG) --version)
+else
+ifeq ($(BUILD_OS),$(OS))
+LLVM_CONFIG := $(build_bindir)/llvm-config$(HOST_EXE)
 else
-LLVM_CONFIG=$(build_bindir)/llvm-config$(EXE)
+LLVM_CONFIG := $(build_bindir)/llvm-config-host$(HOST_EXE)
```

Comment:
In the spirit of "this does too many things at once," how about taking out the llvm-config-host change since it's not at all necessary or interdependent on out-of-tree builds. I get that you're trying to work towards arm cross, but one thing at a time.